packages/eslint-plugin/src/rules/no-invalid-void-type.ts

```diff
@@ -127,6 +127,35 @@ export default util.createRule<[Options], MessageIds>({
       }
     }
 
+    /**
+     * @brief checks if the generic type parameter extends void
+     */
+    function checkExtendedVoid(
+      node: TSESTree.TSVoidKeyword,
+      parentNode: TSESTree.TSTypeParameter,
+    ): void {
+      if (
+        parentNode.constraint?.type !== AST_NODE_TYPES.TSVoidKeyword ||
+        JSON.stringify(parentNode.default?.range) !== JSON.stringify(node.range)
+      ) {
+        return;
+      }
+
+      context.report({
+        messageId: 'invalidVoidNotReturnOrGeneric',
+        node: parentNode.constraint,
+      });
+    }
+
+    /**
+     * @brief checks if the type of given argument is TSESTree.TSTypeParameter
+     */
+    function isTSTypeParameter(
+      node: TSESTree.Node,
+    ): node is TSESTree.TSTypeParameter {
+      return node.type === AST_NODE_TYPES.TSTypeParameter;
+    }
```

Comment:
Nit: this `isTSTypeParameter` function is only used in one place. There's no need to make an extra function for it. You can inline `node.parent.type === AST_NODE_TYPES.TSTypeParameter` where this would be called, below.

Reply:
Thank you!
I deleted the function and inlined it, so please check the below:
https://github.com/typescript-eslint/typescript-eslint/pull/5671/commits/4b7ad4127dae4c6f05623ae86eac77c157bcee36